ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -1038,22 +1107,39 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
       onMount();
     }, []);
 
-    // handle the route /nih-callback?token=<token>
-    // handle the route /ras-callback?code=<code>
     const query = useQuery();
+
+    // handle the route /nih-callback?token=<token>
     const token = query.get('token');
-    const code = query.get('code');
     useEffect(() => {
       if (token) {
         handleTerraShibbolethCallback(token, spinnerProps, reload);
       }
     }, [token]);
+
+    // handle the route /ras-callback?code=<code>
+    const code = query.get('code');
     useEffect(() => {
       if (code) {
         handleRasCallback(code, spinnerProps, reload);
       }
     }, [code]);
 
+    // handle the different page modes of Data Access Requirements
+    const [pageMode, setPageMode] = useState(DARPageMode.INITIAL_REGISTRATION);
+    const pageModeParam = query.get('pageMode');
+    useEffect(() => {
+      if (
+        environment.mergedAccessRenewal &&
+        pageModeParam &&
+        Object.values(DARPageMode).includes(
+          pageModeParam as unknown as DARPageMode
```

Comment:
`as unknown as` is new. Can you explain this a little bit?

Reply:
Enums are hard to work with in Typescript, and this is one way to coerce the correct type.  But I think I've found a better way now - updated.